frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/AddDataModal/Panels/CSVUpload.tsx

```diff
@@ -0,0 +1,299 @@
+import cx from "classnames";
+import {
+  type ChangeEvent,
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { type FileRejection, useDropzone } from "react-dropzone";
+import { c, t } from "ttag";
+
+import { getComposedDragProps } from "metabase/collections/components/CollectionContent/utils";
+import { useGetDefaultCollectionId } from "metabase/collections/hooks";
+import {
+  type CollectionPickerItem,
+  CollectionPickerModal,
+} from "metabase/common/components/CollectionPicker";
+import { UploadInput } from "metabase/components/upload";
+import CollectionName from "metabase/containers/CollectionName";
+import { useDispatch } from "metabase/lib/redux";
+import {
+  MAX_UPLOAD_SIZE,
+  MAX_UPLOAD_STRING,
+  uploadFile,
+} from "metabase/redux/uploads";
+import { Box, Button, Center, Group, Icon, Stack, Text } from "metabase/ui";
+import { UploadMode } from "metabase-types/store/upload";
+
+import S from "../AddDataModal.module.css";
+
+import IconCSV from "./icons/csv.svg?component";
+import IconCSVWarning from "./icons/csv_warning.svg?component";
+
+type UploadState = {
+  file: File | null;
+  error: string | null;
+};
+
+export const CSVUpload = ({
+  onCloseAddDataModal,
+}: {
+  onCloseAddDataModal: () => void;
+}) => {
+  const dispatch = useDispatch();
+  const initialCollectionId = useGetDefaultCollectionId() ?? "root";
+
+  const [uploadState, setUploadState] = useState<UploadState>({
+    file: null,
+    error: null,
+  });
+
+  const [isCollectionPickerOpen, setIsCollectionPickerOpen] = useState(false);
```

Comment:
I think we're trying to standardize on the `useDisclosure` hook for this sort of thing

Reply:
I wasn't aware, but I fully support it.
Even better 🎉 

https://github.com/metabase/metabase/pull/59008/commits/06fc685fef7c80905349f7e9c4b552146f80b63a